app/view/grid/ItemDeleter.js

```diff
@@ -0,0 +1,47 @@
+Ext.define('CpsiMapview.view.grid.ItemDeleter',
+    {
+        extend: 'Ext.grid.column.Action',
+        xtype: 'cmv_itemdeleter',
+        width: 55,
+        requiredRole: this.deleteItemRequiredRole,
+        menuDisabled: true, // disable the column header menu containing sort/hide options
+        iconCls: 'x-fa fa-delete',
+        ignoreConfirmation: false,
+        isActionColumn: true,
+        hideable: false, // prevent the user from hiding this column
+        draggable: false,
+        tooltip: 'Click here to remove the record',
+
+        /**
+            Possible bug when a row editor plugin is added to the same grid as the ItemDeleter?
+
+            column.getEditor().hide() called - and there is no Editor for the ItemDeleter
+
+            http://docs.sencha.com/extjs/4.2.3/source/Column3.html#Ext-grid-column-Column-method-getEditor
+
+            EDIT: by LEO 20160404 - a fake control returned by getEditor() creates more issues, while defining a real editor seems to work
+            */
+        editor: {
+            xtype: 'label',
+            text: ''
+        },
+
+        itemIsDisabled: function (/*view, rowIndex, colIndex, item, record*/) {
```

Comment:
I'm not sure I understand the purpose of this function?

Reply:
basically it replaces the "isDisabled" configuration parameter, according to sencha docs

https://docs.sencha.com/extjs/6.7.0/classic/Ext.grid.column.Action.html#cfg-isDisabled

Reply:
Ok but `itemIsDisabled` is a custom function that is called in `isActionDisabled`, but then both code paths returns `this.disabled`? I don't see when this would ever return a different value?
If it is to be overriden then it should be empty and commented as such.